sklearn/metrics/scorer.py

```diff
@@ -66,6 +68,52 @@ def __repr__(self):
                    "" if self._sign > 0 else ", greater_is_better=False",
                    self._factory_args(), kwargs_string))
 
+    def _get_labels_from_estimator(self, estimator, target_type):
+        """Check if estimator has a classes argument and if scorer has a
+        labels argument and return new kwargs for scorer if label not already
+        set in make score initialization"""
+        est_has_classes = hasattr(estimator, "classes_")
+
+        # if nothing is to be copied
+        if self._copy_classes is None:
+            return self._kwargs
+
+        labels_atrr_name = self._copy_classes
+
+        if labels_atrr_name != 'labels':
+            if labels_atrr_name not in signature(
+                                            self._score_func).parameters:
+                raise ValueError("the scorer doesn't have %s as a parameter,"
+                                 "as passed in copy_classes parameter" %
+                                 labels_atrr_name)
+
+        scorer_has_labels = labels_atrr_name in signature(
```

Comment:
agree. i think the classes check was also not required if we do above. i've simplified it.